monai/data/wsi_reader.py

```diff
@@ -0,0 +1,428 @@
+            size: (height, width) tuple giving the patch size at the given level (`level`).
+                If None, it is set to the full image size at the given level.
+            level: the level number. Defaults to 0
+            dtype: the data type of output image
+            mode: the output image mode, 'RGB' or 'RGBA'
+
+        Returns:
+            a tuples, where the first element is an image patch [CxHxW] or stack of patches,
+                and second element is a dictionary of metadata
+        """
+        patch_list: List = []
+        metadata = {}
+        # CuImage object is iterable, so ensure_tuple won't work on single object
+        if not isinstance(wsi, List):
+            wsi = [wsi]
+        for each_wsi in ensure_tuple(wsi):
+            # Verify magnification level
+            if level is None:
+                level = self.level
+            max_level = self.get_level_count(each_wsi) - 1
+            if level > max_level:
+                raise ValueError(f"The maximum level of this image is {max_level} while level={level} is requested)!")
+
+            # Verify location
+            if location is None:
+                location = (0, 0)
+            wsi_size = self.get_size(each_wsi, level)
+            if location[0] > wsi_size[0] or location[1] > wsi_size[1]:
+                raise ValueError(f"Location is outside of the image: location={location}, image size={wsi_size}")
+
+            # Verify size
+            if size is None:
+                if location != (0, 0):
+                    raise ValueError("Patch size should be defined to exctract patches.")
+                size = self.get_size(each_wsi, level)
+            else:
+                if size[0] <= 0 or size[1] <= 0:
+                    raise ValueError(f"Patch size should be greater than zero, provided: patch size = {size}")
+
+            # Extract a patch or the entire image
+            patch = self.get_patch(each_wsi, location=location, size=size, level=level, dtype=dtype, mode=mode)
+
+            # Verify patch image
+            patch = self.verify_output(patch, mode)
+
+            # Create a list of patches
+            patch_list.append(patch)
+
+            # Set patch-related metadata
+            each_meta = self.get_metadata(patch=patch, location=location, size=size, level=level)
+            metadata.update(each_meta)
+
+        return _stack_images(patch_list, metadata), metadata
+
+    def verify_suffix(self, filename: Union[Sequence[PathLike], PathLike]) -> bool:
+        """
+        Verify whether the specified file or files format is supported by WSI reader.
+
+        The list of supported suffixes are read from `self.supported_formats`.
+
+        Args:
+            filename: filename or a list of filenames to read.
+
+        """
+        return is_supported_format(filename, self.supported_formats)
+
+    def verify_output(self, patch: np.ndarray, mode: str):
```

Comment:
I feel we don't need this function because it's simple and only used once, maybe just put this check in `get_data()` directly?

Thanks.